### PR TITLE
feat(calendar): add timeframe selection for requests

### PR DESCRIPTION
**Pull request by PatrickDohn**
#### Summary

_Enumerate and describe your changes_

- add timeframe selection for sick and vacation single day requests.

#### Checklist

_Double check list and complete these actions before merge_

- [x] PR title is in format `TYPE(scope): description`
- [x] Link PR to Issue (if any issues are related)

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -96,11 +98,9 @@ const onResponse = async (response, isAtMessage = false) => {
         if (validation.isValid) {
           say(`Hello <@${event.user}>! ${validation.reason}`);
         } else {
-          await GoogleSheetUtil.setTodayAsSickDay(slackName);
-          await GoogleCalendarUtil.createSickEvent(slackName);
-          say(`Hello <@${event.user}>! Sick day marked. Hope you feel better! 🤒`);
+          const blocks = slackMessageUtil.selectTimeFrame('sick');
```
Comment (davidholyko) [inline src/app.js:101]: I think we  have enough use cases now that  these can be variables in the `constants` file

```
const VACATION = 'vacation'
const SICK = 'sick'
```

File: src/app.js [hunk 2]
```diff
@@ -223,6 +222,50 @@ app.action(SELECT_PAYMO_TASK_ID, async (response) => {
...
+  await response.ack();
+  const { payload, body, client } = response;
+  const { chat } = client;
+  const { user, state } = body;
+  const slackResponder = await UserUtil.createUser(user.id);
+  const channel = slackResponder.slackUser.slackId;
+  const slackName = slackResponder.slackUser.slackName;
+  const request = payload.block_id;
+
+  if (request === 'sick') {
```
Comment (davidholyko) [inline src/app.js:235]: this is pretty redundant...

try {
 // ...
  let blocks = ''
  if (sick) {
    blocks = `Hello ${slackName}! Sick day marked hope you feel better🤒`
  } 
  if (vacation) {
    block = '...'
  }
  // ...
}

File: src/constants.js [hunk 3]
```diff
@@ -54,6 +54,7 @@ const SLACK_INTERACTIVE_MESSAGE_ACTION_IDS = {
   SELECT_START_DATE: 'select_start_date',
   SELECT_END_DATE: 'select_end_date',
   SUBMIT_DATES: 'submit_dates',
+  SELECT_TIME_FRAME: 'select_time_frame',
```
Comment (davidholyko) [inline src/constants.js:57]: Maybe SELECT_DATE_TIME_FRAME. this refers to a time frame for a single day

File: src/utils/googleCalendarUtil.js [hunk 4]
```diff
@@ -58,20 +58,20 @@ class GoogleCalendarUtil {
...
 
-  async createTodayEvent(slackName, eventType, startDate, endDate) {
+  async createTodayEvent({ slackName, eventType, selectedTimeFrame, startDate, endDate }) {
     const today = moment().format('YYYY-MM-DD');
     const event = {
-      description: 'This event is brought to you by hrbot.',
+      description: `${selectedTimeFrame} \nThis event is brought to you by hrbot.`,
       start: {
-        date: startDate || today,
+        date: today || startDate,
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:70]: this is saying whatever vacation you enter in the vacation date picker will alway bee today

File: src/utils/googleCalendarUtil.js [hunk 5]
```diff
@@ -58,20 +58,20 @@ class GoogleCalendarUtil {
...
+  async createSickEvent(slackName, selectedTimeFrame) {
+    await this.createTodayEvent({ slackName, eventType: 'sick', selectedTimeFrame });
   }
 
-  async createTodayEvent(slackName, eventType, startDate, endDate) {
+  async createTodayEvent({ slackName, eventType, selectedTimeFrame, startDate, endDate }) {
     const today = moment().format('YYYY-MM-DD');
     const event = {
-      description: 'This event is brought to you by hrbot.',
+      description: `${selectedTimeFrame} \nThis event is brought to you by hrbot.`,
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:68]: this  might end up saying 

undefined 
This event is brought to you by hrbot.

for vacation days in the future

File: src/app.js (hunk 1 above)
Comment (davidholyko) [reply src/app.js:101]: are replace all  instances of 'sick' and 'vacation' with the variables

File: src/constants.js [hunk 6]
```diff
@@ -54,6 +57,7 @@ const SLACK_INTERACTIVE_MESSAGE_ACTION_IDS = {
   SELECT_START_DATE: 'select_start_date',
   SELECT_END_DATE: 'select_end_date',
   SUBMIT_DATES: 'submit_dates',
+  SELECT_DATE_TIME_FRAME: 'select_date_time_frame',
```
Comment (davidholyko) [inline src/constants.js:60]: move up one line

File: src/app.js [hunk 7]
```diff
@@ -16,13 +17,15 @@ const { version } = require('../package.json');
...
 const googleCalendarUtil = require('./utils/googleCalendarUtil');
+const slackMessageUtil = require('./utils/slackMessageUtil');
 
 const {
   INPUT_PAYMO_API_KEY,
   SELECT_PAYMO_TASK_ID,
   SELECT_START_DATE,
   SELECT_END_DATE,
   SUBMIT_DATES,
+  SELECT_DATE_TIME_FRAME,
```
Comment (davidholyko) [inline src/app.js:28]: move up one line

File: src/app.js [hunk 8]
```diff
@@ -223,6 +223,53 @@ app.action(SELECT_PAYMO_TASK_ID, async (response) => {
...
+    let message = null;
+    const value = state.values;
+    const key = Object.keys(value);
+    const selectedTimeFrame = value[key].select_time_frame.selected_option.text.text;
+    const events = await GoogleCalendarUtil.getAllSingleDayEvents();
+    const validation = GoogleCalendarUtil.validateUserEvent(slackName, events);
+
+    // if you already have an event today, send message saying that new request is not valid
+    if (!validation.isValid) {
+      const message = `Hello <@${channel}>! ${validation.reason}`;
```
Comment (davidholyko) [inline src/app.js:247]: just

```
message = /* ... */
```

no const

File: src/constants.js [hunk 9]
```diff
@@ -18,6 +18,9 @@ const RESOURCES = {
   },
 };
 
+const VACATION = 'vacation';
```
Comment (davidholyko) [inline src/constants.js:21]: Maybe wrapping in name space CALENDAR_EVENT_TYPE

```
const CALENDAR_EVENT_TYPE = {
  // ...
}
```

File: src/utils/googleCalendarUtil.js [hunk 10]
```diff
@@ -19,8 +19,8 @@ class GoogleCalendarUtil {
     });
   }
 
-  async createVacationEvent(slackName) {
-    await this.createTodayEvent(slackName, 'vacation');
+  async createVacationEvent(slackName, selectedTimeFrame) {
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:22]: maybe `dateRange` makes more sense than `selectedTimeFrame`? Less characters

File: src/utils/slackMessageUtil.js [hunk 11]
```diff
@@ -15,6 +15,7 @@ const {
   SELECT_START_DATE,
   SELECT_END_DATE,
   SUBMIT_DATES,
+  SELECT_DATE_TIME_FRAME,
```
Comment (davidholyko) [inline src/utils/slackMessageUtil.js:18]: move up a line

File: src/utils/slackMessageUtil.js [hunk 12]
```diff
@@ -278,6 +279,47 @@ class SlackMessageUtil {
     ];
     return blocks;
   }
+
+  selectTimeFrame(request) {
```
Comment (davidholyko) [inline src/utils/slackMessageUtil.js:283]: `selectDateRange`? Usually when I think of time, its hour. Date is days

File: src/utils/slackMessageUtil.js [hunk 13]
```diff
@@ -278,6 +279,47 @@ class SlackMessageUtil {
...
+    const blocks = [
+      {
+        type: 'section',
+        block_id: request,
```
Comment (davidholyko) [inline src/utils/slackMessageUtil.js:287]: maybe rename `request` with `eventType`. Request usually refers to a network request

File: src/utils/googleCalendarUtil.js (hunk 10 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js:22]: agreed

File: src/utils/slackMessageUtil.js (hunk 12 above)
Comment (PatrickDohn) [reply src/utils/slackMessageUtil.js:283]: This function refers to the to whole day or half day though.

Comment (davidholyko) [reply src/utils/slackMessageUtil.js:283]: oh, true I was  thinking about something else

File: src/utils/googleCalendarUtil.js (hunk 10 above)
Comment (davidholyko) [reply src/utils/googleCalendarUtil.js:22]: actually its fine as `selectedTimeFrame`

File: src/utils/slackMessageUtil.js (hunk 13 above)
Comment (PatrickDohn) [reply src/utils/slackMessageUtil.js]: Fixed

File: src/utils/slackMessageUtil.js (hunk 11 above)
Comment (PatrickDohn) [reply src/utils/slackMessageUtil.js:17]: Fixed

File: src/utils/googleCalendarUtil.js (hunk 10 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js]: Fixed

File: src/constants.js (hunk 9 above)
Comment (PatrickDohn) [reply src/constants.js]: Fixed

File: src/app.js (hunk 8 above)
Comment (PatrickDohn) [reply src/app.js]: Fixed

File: src/app.js (hunk 7 above)
Comment (PatrickDohn) [reply src/app.js:27]: Fixed

File: src/constants.js (hunk 6 above)
Comment (PatrickDohn) [reply src/constants.js:61]: Fixed

**Review by davidholyko — APPROVED** [review]

**Pull request merged**